File: Ejercicio4/ordensup.py
def suma(x,y):
    return x + y

def resta(x,y):
    return x - y

def mult(x,y):
    return x * y

def div(x,y):
    return x / y

def calcu(a,b,func):
    res = func(a,b)
    print(res)

bo = True
while(bo):
    print('--------- Calculadora ---------')
    a = int(input("a: "))
    b = int(input("b: "))
    op = input("Operación(suma, resta, mult, div): ")
    if (op == 'suma'):
        calcu(a,b,suma)
    elif (op == 'resta'):
        calcu(a,b,resta)
    elif (op == 'mult'):
        calcu(a,b,mult)
    elif (op == 'div'):
        calcu(a,b,div)
    else:
        print("operación no reconocida")
    cont = input("Continuar? s/n \n")
    if(cont != "s"):
        bo = False

File: Ejercicio4/ordensup.scala
println("---------- Calculadora ----------")
def suma(a:Int,b:Int):Int=a+b
def resta(a:Int,b:Int):Int=a-b
def mult(a:Int,b:Int):Int=a*b
def div(a:Int,b:Int):Int=a/b
def mod(a:Int,b:Int):Int=a%b

def xay(x:Int, y:Int, funcion:(Int, Int) => Int):Int = funcion(x,y)

def calcu(x:Int, y:Int, operacion:String):Int={
    operacion match {
        case "suma"=>xay(x,y,suma)
        case "resta"=>xay(x,y,resta)
        case "mult"=>xay(x,y,mult)
        case "div"=>xay(x,y,div)
    }
}

var bool : Boolean = true;
while(bool == true){
    print("a:")
    val a = scala.io.StdIn.readInt()
    print("b:")
    val b = scala.io.StdIn.readInt()
    print("Operacion (suma, resta, mult o div):")
    val op = scala.io.StdIn.readLine()
    var res = 0
    if(op == "suma" || op == "resta" || op == "mult" || op == "div" ) {
        res = calcu(a,b,op)
        if(op == "div"){
            var m = xay(a,b,mod) / b.toFloat
            var r = res.toFloat + m
            println(r)
        }
        else println(res)
    }
    else println("Operación no reconocida")
    println("Continuar? s/n")
    var co = scala.io.StdIn.readLine()
    if(co != "s") bool = false
}